### ISSUE TITLE
Backup permissions are not persisted

### ISSUE THREAD

**Issue by scholtzm** [honestbleeps/Reddit-Enhancement-Suite#4358]
- My browser: Firefox 54 on Windows 8.1
- My RES version: v5.8.1
- Am I in reddit beta (yes/no): no

*What's the problem?*
RES will ask for permission to backup to **Dropbox** after every browser restart.

Steps to reproduce:
1. Click the "Backup - Dropbox" button.
2. Authorize RES to use Dropbox.
3. Backup is completed.
4. Click the button again, backup is completed with additional authorization.
5. Restart browser and try to backup to Dropbox - browser will need to be reauthorized.

Seems like every backup eventually goes through: [`chrome.identity.launchWebAuthFlow`](https://github.com/honestbleeps/Reddit-Enhancement-Suite/blob/master/firefox/background.entry.js#L19). My assumption is that this method either opens the authorization window or just provides the `responseUrl` if it has been requested previously. However if the user restarts browser, the value is lost and the authorization request needs to be completed again.

*Are there any related posts in https://github.com/honestbleeps/Reddit-Enhancement-Suite/issues or /r/RESissues?*
https://www.reddit.com/r/RESissues/comments/6nrk6e/res_continually_asking_for_permission_to_backup/

*What other browser extensions are installed?*
ABP, React Dev Tools

**Comment by erikdesjardins** [comment]
`launchWebAuthFlow` runs the auth flow in a hidden window, it doesn't just remember the one response. (Works on my machine<sup>TM</sup>, same browser version.)

When you restart the browser, are you also logged out of dropbox.com? If so, try to fix that and this issue should disappear.

**Comment by scholtzm** [comment]
@erikdesjardins 

Ok, everything "clicked" after your response.

I haven't logged into dropbox.com in a very long time. When I clicked the Dropbox button, it launched **interactive** auth flow. This opens login form without "Remember me" checkbox, hence the need to re-authorize RES after every browser restart. This wasn't immediatelly obvious to me and seems like several other people ran into this issue.

To fix this, I just went to dropbox.com and logged in. This time, after clicking the Dropbox button, non-interactive auth flow worked and my data was backed up right away.

To me, it looks like the interactive auth flow is only usable for a one time backup.

**Comment by jrittenh** [comment]
I have a similar issue, but using Google Chrome.  For me, it appears to be tied to the multiple Google profiles I have logged in at once.  Every "first" access to reddit prompts for login, and I then have to select which account I would like to use.  No additional auth is necessary if I reuse an account, but I have to choose the account every time the browser's session ends.

**Comment by Setmaster** [comment]
Same issue @jrittenh

**Comment by shiruken** [comment]
I am also experiencing the same issue as @jrittnh and can confirm it appears to be associated with multiple logged in Google accounts.

**Comment by gabrielsoldani** [comment]
@jrittenh @Setmaster @shiruken This is a different issue, see #4376.

**Comment by corylulu** [comment]
Auto-backup to Dropbox forgets my home PC and gives me a prompt once a day, but my work PC it always remembers. Also this isn't between sessions or anything, my home PC never turns off and always has browsers opened.

**Comment by larsjohnsen** [comment]
@corylulu You may have to logged into the Dropbox.com to avoid those daily prompts.

**Comment by corylulu** [comment]
@larsjohnsen I am logged into dropbox.com, though. Also tried cleaning up my devices list on their site as well. Still got the prompt though.

**Comment by andystevens91** [comment]
Same thing happens for me using OneDrive, using Firefox 56.0b8 and RES 5.8.4.

**Comment by chrisblakley** [comment]
Same issue here. I logged into dropbox.com to confirm all the apps and security settings, but I'm still getting it.

**Comment by ELLIOTTCABLE** [comment]
I've tried logging out, and back in, on Dropbox.com; no dice over here.

Same problem.

**Comment by theinfra** [comment]
Chrome Version 65.0.3325.181
RES 5.10.3

Was having this issue for a couple of days and ever since I logged into the Dropbox website (like suggested [here ](https://github.com/honestbleeps/Reddit-Enhancement-Suite/issues/4358#issuecomment-322897182)by @larsjohnsen) there hasn't been a repeat prompt for permissions since.

I was using Google Drive as a backup service but since I use Reddit on 2 different PC's with different settings for each, I decided to switch to Dropbox on one. I haven't used Dropbox for ages,  so 100% I wasn't logged into the website. I just authorized the API prompt on http://redditenhancementsuite.com and the prompts where very frequent, sometimes on the same browser session without closing. I opened session on dropbox.com and the issue got solved.

**Comment by corylulu** [comment]
Yeah, the problem never went away for me... I just moved it to Google Drive and it worked fine. I did everything I could think of to stop it. No luck.

**Comment by NeckBeardPrince** [comment]
Alright, so this is still an issue.

**Comment by offbyone** [comment]
3 years in the future, it's a _new_ issue for me.

**Comment by arduoi** [comment]
Still happening
